server/src/grpc/remote_engine_service/mod.rs

```diff
@@ -161,10 +164,12 @@ impl<Q: QueryExecutor + 'static> RemoteEngineService for RemoteEngineServiceImpl
                                 header: Some(error::build_err_header(e)),
                                 ..Default::default()
                             },
-                            Ok(CompressOutput { payload, .. }) => ReadResponse {
+                            Ok(CompressOutput { payload, method }) => ReadResponse {
                                 header: Some(build_ok_header()),
-                                version: RemoteEngineVersion::ArrowIPCWithZstd.as_u32(),
-                                rows: payload,
+                                output: Some(Arrow(ArrowPayload {
+                                    record_batches: vec![payload],
+                                    compression: method as i32,
```

Comment:
The type of `method` is not the same as the type of `compression` field, and maybe we need an explicit conversion between them.